src/azure-cli-core/azure/cli/core/tests/test_util.py

```diff
@@ -121,6 +121,27 @@ def _run_test(length, force_lower):
         # Test force_lower
         _run_test(16, True)
 
+    def test_proxy_resource_parse(self):
+        mock_proxy_resource_id = "/subscriptions/0000/resourceGroups/clirg/" \
+                                 "providers/Microsoft.Network/privateEndpoints/cli/" \
+                                 "privateLinkServiceConnections/child_name_1/child_type_2/child_name_2"
+        result = parse_proxy_resource_id(mock_proxy_resource_id)
```

Comment:
@qianwens Hi. This is the usage scenario. It's impossible to have same key. This is a similar pattern with `parse_resource_id`.

Reply:
for client who use this function, how do they know they need to get value of "chile_type" by key "child_type_1"

---
In reply to: [386765720](https://github.com/Azure/azure-cli/pull/12403#discussion_r386765720) [](ancestors = 386765720)

Reply:
Hi. Sounds good to have a sub sequence list. However, as I said, I follow the structure of msrestazure's `parse_resource_id`, which is developed two years ago and commonly used in azure-cli. I think it's more intuitive to keep them consistent and easier for users.
https://github.com/Azure/msrestazure-for-python/blob/0a07586816351e3b7abd60a55a8b736e007fb7a8/msrestazure/tools.py#L106

Reply:
The original design of parse_resource_id seems not intuitive and not common design for a method, we do not need to follow that method unless they are in the same module. There is no need to convert a sequence list to a dictionary with the naming conversion.  

---
In reply to: [386835902](https://github.com/Azure/azure-cli/pull/12403#discussion_r386835902) [](ancestors = 386835902)